[CORE-8870]  Provide more UUID information

michael-redpanda opened the pull request:
<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

This adds:
* `/v1/cluster/metrics_uuid` - returns the UUID reported by the metrics telemetry system
* Adds `storage_uuid` to the metrics telemetry report

Fixes: CORE-8870
Fixes: CORE-8904

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [X] v24.3.x
- [X] v24.2.x
- [X] v24.1.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

* None

michael-redpanda committed:
admin: Added /v1/cluster/metrics_uuid

Endpoint returns the UUID reported by the metrics Endpoint

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
cluster: Reporting storage UUID in metrics data

Signed-off-by: Michael Boquard <michael@redpanda.com>

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--035b5d39-392c-49cb-9dcf-e556e3dbf7b1-->
<!--TEST RESULTS-->
<!--f2257c0d-0f20-4496-a31c-6c1295b61145-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61166">build#61166</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compaction_recovery_test.CompactionRecoveryUpgradeTest.test_index_recovery_after_upgrade | ducktape | https://buildkite.com/redpanda/redpanda/builds/61166#019498e9-1ad6-4e2c-b8d6-f5212cb4838a | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61246">build#61246</a></summary>


| test_id                                                                         | test_kind   | job_url                                                                                   | test_status   | passed   |
|:--------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.scaling_up_test.ScalingUpTest.test_scaling_up_with_recovered_topic | ducktape    | https://buildkite.com/redpanda/redpanda/builds/61246#0194aa03-bdf3-4d7b-ba01-23a619ebc543 | FLAKY         | 1/2      |
</details>
<!--/body-->

pgellert commented on src/v/redpanda/admin/api-doc/cluster.json:
```diff
@@ -366,4 +397,4 @@
         }
     }
-}
+}
```
nit: the newline at the end of the file was removed, which is probably unintended

pgellert reviewed (dismissed):
lgtm

dotnwat reviewed (commented):
lgtm. could probably benefit from a short test that pokes the new endpoint so we make sure it stays alive.

BenPope reviewed (commented):
LGTM, aside from the other comments

michael-redpanda commented:
Force push [a026f19](https://github.com/redpanda-data/redpanda/commit/a026f190d25db07266adbc28460f88a70f2f1d35):

* Added test
* `controller_stm` is a single-sharded service to calling `.local()` on it can fail.  Changes to invoking on shard 0

Force push [bafbc9f](https://github.com/redpanda-data/redpanda/commit/bafbc9fd8ed195d95cf4f76fb83d20d3074431db):

* Fixed whitespace issue in json file

michael-redpanda commented:
Force push [d333e4c](https://github.com/redpanda-data/redpanda/commit/d333e4cc19693e18acb7e36507c91ae6c66b4ce1):

* Fixed python lint

michael-redpanda commented:
Force push [92217ca](https://github.com/redpanda-data/redpanda/commit/92217caf8aa2528e4e2e439d1219523b99b51cdb):

* Fixed user-after-free lint error with capturing `this` in a coroutine lambda

BenPope commented on tests/rptest/services/admin.py:
```diff
@@ -1451,6 +1451,33 @@ def get_cluster_uuid(self, node=None):
+        None
+            If the endpoint returns a 404 status code.
+        """
+        try:
```
Should there be an `await_stable_leader` somewhere?

michael-redpanda replied:
yeah that's probably a good idea

BenPope replied:
> yeah that's probably a good idea

It should probably also go to a leader if no node is specified, right?

michael-redpanda replied:
It appears that information gets replicated so it shouldn't matter

michael-redpanda commented:
Force push [9df3a90](https://github.com/redpanda-data/redpanda/commit/9df3a906833f2adb0dfe520ebce15f425a96a70a):

* Awaiting controller leadership in get_metrics_uuid test

michael-redpanda committed:
rpk: Added metrics UUID to debug bundle

Signed-off-by: Michael Boquard <michael@redpanda.com>

r-vasquez reviewed (commented):
rpk changes LGTM. I'll leave the approval to Core. Thanks!

vbotbuildovich commented:
/backport v24.3.x

vbotbuildovich commented:
/backport v24.2.x

vbotbuildovich commented:
/backport v24.1.x

vbotbuildovich commented:
Failed to create a backport PR to v24.1.x branch. I tried:

```
git remote add upstream https://github.com/redpanda-data/redpanda.git
git fetch --all
git checkout -b backport-pr-24929-v24.1.x-600 remotes/upstream/v24.1.x
git cherry-pick -x fbc117bec9 9df3a90683 62dea8154e
```

[Workflow run logs](https://github.com/redpanda-data/redpanda/actions/runs/13010996144).

vbotbuildovich commented:
Failed to create a backport PR to v24.2.x branch. I tried:

```
git remote add upstream https://github.com/redpanda-data/redpanda.git
git fetch --all
git checkout -b backport-pr-24929-v24.2.x-481 remotes/upstream/v24.2.x
git cherry-pick -x fbc117bec9 9df3a90683 62dea8154e
```

[Workflow run logs](https://github.com/redpanda-data/redpanda/actions/runs/13010996052).
